xtask/src/main.rs

```diff
@@ -0,0 +1,177 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+type DynError = Box<dyn std::error::Error>;
+
+fn main() {
+    if let Err(e) = try_main() {
+        eprintln!("{}", e);
+        std::process::exit(-1);
+    }
+}
+
+fn try_main() -> Result<(), DynError> {
+    let task = env::args().nth(1);
+    match task.as_deref() {
+        Some("build") => build()?,
+        _ => print_help(),
+    }
+    Ok(())
+}
+
+fn print_help() {
+    eprintln!(
+        "Tasks:
+
+wasm            build wasm using wasm32-wasi target
+components      build components using wasm-tools
+"
+    )
+}
+
+struct Guest {
+    name: String,
+    path: PathBuf,
+}
+
+fn guests() -> Vec<Guest> {
+    let path_guests = project_root().join("guests/");
+    let paths = fs::read_dir(path_guests.clone()).unwrap();
+    return paths
+        .map(|p| p.unwrap().path())
+        .filter(|p| p.is_dir())
+        .map(|p| {
+            let path = p.clone();
+            let name = p
```

Comment:
not quite following whats happening here. Mind just tossing a small comment in the code here.